tests/phpunit/tests/fonts/font-library/wpFontsDir.php

```diff
@@ -69,4 +81,61 @@ function set_new_values( $defaults ) {
 
 		$this->assertSame( static::$dir_defaults, $font_dir, 'The wp_get_font_dir() method should return the default values.' );
 	}
+
+	public function test_should_create_fonts_dir_in_uploads_when_fails_in_wp_content() {
+		// Set the expected results.
+		$upload_dir = wp_upload_dir();
+		$path       = path_join( $upload_dir['basedir'], 'fonts' );
+		$expected   = array(
+			'path'    => $path,
+			'url'     => $upload_dir['baseurl'] . '/fonts',
+			'subdir'  => '',
+			'basedir' => $path,
+			'baseurl' => $upload_dir['baseurl'] . '/fonts',
+			'error'   => false,
+		);
+
+		$this->fake_no_new_directories_in_wp_content();
+		$this->assertFileExists( self::$fake_fonts_file );
+
+		$font_dir = wp_get_font_dir();
+
+		$this->assertDirectoryDoesNotExist( path_join( WP_CONTENT_DIR, 'fonts' ), 'The `wp-content/fonts` directory should not exist.' );
+		$this->assertDirectoryExists( $font_dir['path'], 'The `uploads/fonts` directory should exist.' );
+		$this->assertSame( $expected, $font_dir, 'The `fonts` directory should be a subdir in the `uploads` directory.' );
+	}
+
+	private function remove_fonts_directory() {
+		$directories = array(
+			path_join( WP_CONTENT_DIR, 'fonts' ),
+			path_join( WP_CONTENT_DIR, 'uploads/fonts' ),
+			'/custom-path/fonts/my-custom-subdir',
+		);
+
+		foreach ( $directories as $dir ) {
+			if ( ! is_dir( $dir ) ) {
+				continue;
+			}
+
+			$this->rmdir( $dir );
+			@rmdir( $dir );
+		}
+	}
+
+	/**
+	 * A placeholder "fake" file at `wp-content/fonts` triggers `wp_mkdir_p()` to fail into the `file_exists()` bail out, causing `is_dir()` to  return `false`.
+	 * This effectively makes `wp-content/fonts` unwritable as the default `fonts` directory location.
+	 */
+	private function fake_no_new_directories_in_wp_content() {
+		file_put_contents(
+			self::$fake_fonts_file,
+			'fake file'
+		);
+	}
+
+	private function remove_no_new_directories_in_wp_content_fake() {
+		if ( file_exists( self::$fake_fonts_file ) ) {
+			@unlink( self::$fake_fonts_file );
```

Comment:
I don't think we need or should add error suppression.

Reply:
removed here: https://github.com/WordPress/wordpress-develop/pull/6259/commits/f7de278c467ffa43205e32f0605aee6b94d988fd